addons/point_of_sale/tests/test_pos_products_with_tax.py

```diff
@@ -294,6 +296,131 @@ def test_return_order(self):
+            "report_id": tax_report.id,
+            "sequence": 2,
+        }
+        tax_20 = self.env["account.tax.report.line"].create(vals)
+        tax_20_tag_minus = tax_20.tag_ids.filtered("tax_negate")
+        vals = {
+            "sequence": 20,
+            "description": "20%",
+            "name": "20%",
+            "amount": 20,
+            "amount_type": "percent",
+            "type_tax_use": "sale",
+            'price_include': True,
+            "invoice_repartition_line_ids": [
+                (0, 0, {
+                    'factor_percent': 100,
+                    'repartition_type': 'base',
+                    'tag_ids': [(6, 0, base_20.tag_ids.filtered(
+                        lambda t: not t.tax_negate).ids)],
+                }),
+                (0, 0, {
+                    'factor_percent': 100,
+                    'repartition_type': 'tax',
+                    'account_id': self.tax_received_account.id,
+                    'tag_ids': [(6, 0, tax_20.tag_ids.filtered(
+                        lambda t: not t.tax_negate).ids)],
+                }),
+            ]
+        }
+        tax_20_incl = self.env['account.tax'].create(vals)
+        product1 = self.create_product(
+            name='Product 1',
+            category=self.categ_basic,
+            lst_price=12.10,
+            tax_ids=tax_20_incl.ids,
+        )
+        product2 = self.create_product(
+            name='Product 2',
+            category=self.categ_basic,
+            lst_price=6.05,
+            tax_ids=tax_20_incl.ids,
+        )
+        # Open a POS session without creating corresponding moves
+        self.open_new_session()
+        self.env['pos.order'].create_from_ui([self.create_ui_order_data([
+            (product1, 1),
+            (product2, -1),
+        ])])
+
+        # Create an entry
+        with Form(self.env["account.move"].with_context(
+                default_move_type="entry")) as move_form:
+            with move_form.line_ids.new() as line_form:
+                line_form.account_id = self.sale_account
+                # line_form.tax_ids = tax_21_incl
+                line_form.credit = 50.0
+                line_form.tax_ids.add(tax_20_incl)
+            with move_form.line_ids.new() as line_form_2:
+                line_form_2.account_id = self.pos_receivable_account
+                # line_form.tax_ids = tax_21_incl
+                line_form_2.debit = 50.0
+                line_form_2.tax_ids.add(tax_20_incl)
+        move = move_form.save()
+        # Check if tags are not affected by opened POS session
+        self.assertEqual(
+            move.line_ids.filtered(
+                lambda line: line.account_id == self.sale_account).tax_tag_ids,
+            base_20_tag_minus
+        )
+        self.assertEqual(
+            move.line_ids.filtered(
+                lambda line: line.account_id == self.tax_received_account).tax_tag_ids,
+            tax_20_tag_minus
+        )
```

Comment:
shorter and easier to read
```suggestion
        sale_line_tag = move.line_ids.filtered(lambda line: line.account_id == self.sale_account).tax_tag_ids
        receivable_line_tag = move.line_ids.filtered(lambda line: line.account_id == self.tax_received_account).tax_tag_ids
        self.assertEqual(sale_line_tag, base_20_tag_minus)
        self.assertEqual(receivable_line_tag, tax_20_tag_minus)
```

Reply:
I think it would even be better to use the `assertRecordValues` and to test all the lines of the moves.
Like here: (but just test the account, balance and tags) value you are interested in
https://github.com/odoo/odoo/blob/52427b94a99ae76a2a35ef607b46570ec6b5b736/addons/point_of_sale/tests/test_pos_products_with_tax.py#L459-L465